hw/ip/aes/doc/_index.md

```diff
@@ -133,14 +139,25 @@ The AES unit operates as follows:
 1. The initial key and configuration is provided to the AES unit via a set of control and status registers (CSRs) accessible by the processor via TL-UL bus interface.
    Each key register has to be written at least once.
    The order in which the registers are written does not matter.
+1. The processor provides the initialization vector (IV) or initial counter value to the four IV registers via TL-UL bus interface in CBC or CTR mode, respectively.
+   Each IV register has to be written at least once.
+   The order in which the registers are written does not matter.
+   In ECB mode, no IV must be provided.
+   The content of the IV registers is ignored in ECB mode.
 1. The input data is provided to the AES unit via four CSRs.
    Each input register has to be written at least once.
    The order in which the registers are written does not matter.
-1. If new input data is available, or when receiving the start command in the case of manual operation, the AES unit first loads the input data as well as the initial key into the corresponding registers (State and Full Key at the top of the block diagram).
-   Note, if the decryption is performed, the Full Key register is loaded with the value stored in the Decryption Key register.
-1. Once these registers have been loaded, the AES unit starts the encryption/decryption by adding the first round key to the state (all blocks in both data paths are bypassed).
+1. If new input data is available, or when receiving the start command in the case of manual operation the following actions are performed.
+    1. Depending on the cipher mode, the AES unit loads a combination of input data as well as IV into the State register of the cipher core.
+       Note, if the CBC decryption is performed, or if running in CTR mode, the input data is also registered (Data In Prev in the block diagram).
+    2. The initial key is loaded into the Full Key register inside the cipher core.
+       Note, if the ECB/CBC decryption is performed, the Full Key register is loaded with the value stored in the Decryption Key register.
+   Note, for the AES unit to automatically start in CBC/CTR mode, also the IV must be ready.
+   The IV is ready if -- since the last IV update -- the processor has either written all IV registers at least once or none of them.
+   The AES unit will not start the next encryption/decryption with a partially updated IV.
```

Comment:
this could cause SW confusion. Not that it is worth overly complex feedback, but perhaps
it is better to just caveat emptor, SW gets whatever garbage was lingering in IV from before?

Reply:
I have clarified in the doc that in manual mode, we don't care about partially updated IVs (just like partially updated input data registers). Do you think this is sufficient?

Or would you prefer that I changed the HW to not care at all? As pointed out above, I think we want to avoid that people mess around with partially updated IVs.

Reply:
This is another one where DV complexity can come into play, so I recommend that
@rasmus-madsen study this mode and see if we're adding too much DV complexity
for not a lot of security gain.

Reply:
@sjgitty using the openSSlfunctions that Pirmin has implemented I don't think this adds any complexity other than also supplying the IV.

the complexity comes when we want to use a per block verification method as DV would have to calculate intermediate IV's
while not straight forward I don't think this is super complex.
@vogelpi and I had a meeting on this last week and I have a good idea on how to implement this. 
I think the discussion here is not about complexity but if the limited security gain is worth the implementation effort.